jetstream_pt/layers.py

```diff
@@ -59,15 +81,275 @@ def __init__(self, in_features, out_features, bias=None, device=None):
     )
     self.register_buffer("weight_scaler", weight_scaler)
 
-    # if bias:
-    #   self.bias = torch.nn.Parameter(
-    #     torch.zeros((out_features, ),
-    #     dtype=torch.bfloat16, device=device))
-    # else:
-    #   self.register_parameter('bias', None)
+    self.is_symmetric = is_symmetric
+    if not is_symmetric:
+      zero_point = torch.ones(
+          (out_features,), dtype=torch.bfloat16, device=device
+      )
+      self.register_buffer("zero_point", zero_point)
+    else:
+      self.register_buffer("zero_point", None)
+
+    assert not bias, "Quantized Linear doesn't support bias."
+
+    self.n_bit = n_bit
+    # Flag to enable dequantize weight first, then do matmul. Useful for debugging.
+    self.run_fake_quantize = False
+
+  def _load_quantized_weights(self, w_q, scale, zp=None):
+    """
+    Load weights quantized by 'quantize_tensor'.
+    """
+    self.weight, self.weight_scaler, self.zero_point = load_q_weight_helper(
+        w_q, scale, zp, block_size=-1
+    )
+
+  def quantize_weight_from_nn_linear(self, weight):
+    assert weight.dim() == 2, "Expect 2D weight from torch.nn.Linear."
+    assert weight.shape == (
+        self.out_features,
+        self.in_features,
```

Comment:
nit: add expected weight shape in assert error message to help engineer figure out what are correct shape when it's wrong.

Reply:
Makes sense! Done